core/include/detray/propagator/navigator.hpp

```diff
@@ -621,16 +626,79 @@ class navigator {
         if (navigation.trust_level() == navigation::trust_level::e_high or
             navigation.n_candidates() == 1) {
 
+            auto &stepping = propagation._stepping;
+            auto candidate_cache = *navigation.next();
+
             // Update next candidate: If not reachable, 'high trust' is broken
             if (not update_candidate(*navigation.next(), track, det)) {
-                navigation.set_state(navigation::status::e_unknown,
-                                     geometry::barcode{},
-                                     navigation::trust_level::e_no_trust);
-                return;
+
+                // Case 1: When the track overstepped over the overstep
+                // tolerance
+                if (navigation.next()->status ==
```

Comment:
This is curious, if the candidate was not updated successfully, which is the only way to get here (and can happen normally outside of a missed overstepping correction btw) then the intersection status cannot not be ```e_inside```. In fact, any intersection that happens outside of the overstepping tolerance will not be ```e_inside```. Are you sure this case is hit? If not, it will always land in case 2 where the defunct volume grid is queried. That might explain why the segfault is happening even in the straight line navigation.

I would rather check for ```navigation() < track.overstepping_tolerance()```

Reply:
>  any intersection that happens outside of the overstepping tolerance will not be e_inside

Ok I totally misunderstood the intersection code